autode/opt/coordinates/internals.py

```diff
@@ -173,22 +183,12 @@ def _calc_B(self, x: np.ndarray) -> None:
                 "primitive internal coordinates"
             )
 
-        cart_coords = x.reshape((-1, 3))
+        cart_coords = x.ravel()
```

Comment:
@t-young31 `flatten()` seems to always return a copy of the array, but `ravel()` only copies when needed (https://stackoverflow.com/a/28930580). Since `x` should be a flat array anyways, adding `ravel()` seems like a cheap safeguard against wrong shaped arrays (primarily Nx3 coords array).

Reply:
oh cool – thanks for the explanation 👍🏼. agree that _flattening_ makes sense here